### PR TITLE
README: clarify Idris 1 vs 2 situation, and the compilation guide

### PR DESCRIPTION
**Pull request by attila-lendvai**
this is the fruit of my experience as a newcomer to Idris, and trying to compile Idris 1 from source.

### CODE REVIEW

**Comment by juhp** [comment]
I haven't verified all the details but this LGTM anyway.

File: README.md [hunk 1]
```diff
@@ -26,9 +26,18 @@ More information about building Idris from source has been detailed in the [Inst
...
+More information about code generators can be found
+[here](https://idris.readthedocs.io/en/latest/reference/codegen.html).
+
+## Status
+
+This is Idris 1, implemented in Haskell. Idris 1 is not actively
+worked on anymore.
+
+[Idris 2](https://github.com/idris-lang/Idris2) is an early preview of
+the next generation of Idris, implemented in Idris 1.
```
Comment (melted) [inline README.md:40]: Idris 2 is implemented in Idris 2, it was bootstrapped from Idris2-boot, which is the one implemented in Idris 1

Comment (melted) [reply README.md:40]: Also, it's probably less unstable than an "early preview" now. But I don't know how Edwin would characterize it.

**Comment by attila-lendvai** [comment]
ok, i've addressed the concerns by not saying anything about the bootstrap of idris 2. i guess it's more complicated than being discussed in the readme of idris 1... :)

**Review by melted — APPROVED** [review]

**Pull request merged**